# Putting It All Together

Congratulations for reaching this far! Now, we will wrap our workshop by performing exploratory and predictive analytics.

#### Problem: Loan Prediction
Given various independent variable, can you predict whether the loan is approved or not?

#### Variables
![Loan Prediction](images/LoanPrediction.png)

In [37]:
# Import necessary packages for analysis
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

%matplotlib inline

In [38]:
# Load train.csv into pandas DataFrame and name the variable df_train. Directory has already been given below
path_to_train = 'data/LoanPrediction/train.csv'
df_train = pd.read_csv(path_to_train)

In [39]:
# Take a look at the first five entries of df_train
df_train.head()

Loan_ID Gender Married Dependents     Education Self_Employed  \
0  LP001002   Male      No          0      Graduate            No   
1  LP001003   Male     Yes          1      Graduate            No   
2  LP001005   Male     Yes          0      Graduate           Yes   
3  LP001006   Male     Yes          0  Not Graduate            No   
4  LP001008   Male      No          0      Graduate            No   

   ApplicantIncome  CoapplicantIncome  LoanAmount  Loan_Amount_Term  \
0             5849                0.0         NaN             360.0   
1             4583             1508.0       128.0             360.0   
2             3000                0.0        66.0             360.0   
3             2583             2358.0       120.0             360.0   
4             6000                0.0       141.0             360.0   

   Credit_History Property_Area Loan_Status  
0             1.0         Urban           Y  
1             1.0         Rural           N  
2             1.0         Urban           Y  
3             1.0         Urban           Y  
4             1.0         Urban           Y

In [40]:
# Load test.csv into pandas DataFrame and name the variable df_test. Directory has already been given below.
path_to_test = 'data/LoanPrediction/test.csv'
df_test = pd.read_csv(path_to_test)

In [41]:
# Take a look at the first five entries of df_test
df_test.head()

Loan_ID Gender Married Dependents     Education Self_Employed  \
0  LP001015   Male     Yes          0      Graduate            No   
1  LP001022   Male     Yes          1      Graduate            No   
2  LP001031   Male     Yes          2      Graduate            No   
3  LP001035   Male     Yes          2      Graduate            No   
4  LP001051   Male      No          0  Not Graduate            No   

   ApplicantIncome  CoapplicantIncome  LoanAmount  Loan_Amount_Term  \
0             5720                  0       110.0             360.0   
1             3076               1500       126.0             360.0   
2             5000               1800       208.0             360.0   
3             2340               2546       100.0             360.0   
4             3276                  0        78.0             360.0   

   Credit_History Property_Area  
0             1.0         Urban  
1             1.0         Urban  
2             1.0         Urban  
3             NaN         Urban  
4             1.0         Urban

### Exploratory Data Analysis
We will now explore the data found in our train set. In the following code cells, you will be guided on what to do.

In [42]:
# Use .info() to see variable types and other information about the training set
df_train.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 614 entries, 0 to 613
Data columns (total 13 columns):
Loan_ID              614 non-null object
Gender               601 non-null object
Married              611 non-null object
Dependents           599 non-null object
Education            614 non-null object
Self_Employed        582 non-null object
ApplicantIncome      614 non-null int64
CoapplicantIncome    614 non-null float64
LoanAmount           592 non-null float64
Loan_Amount_Term     600 non-null float64
Credit_History       564 non-null float64
Property_Area        614 non-null object
Loan_Status          614 non-null object
dtypes: float64(4), int64(1), object(8)
memory usage: 62.4+ KB


What did you notice from the information above?

1. There are 614 entries.
2. There are missing data for Gender, Married, Dependents, Self_Employed, LoanAmount, Loan_Amount_term, and Credit_History

In [43]:
# Use .describe() to see a numerical summary of the variables in the training set
df_train.describe()

/Users/User/anaconda3/lib/python3.5/site-packages/numpy/lib/function_base.py:3834: RuntimeWarning: Invalid value encountered in percentile
  RuntimeWarning)


ApplicantIncome  CoapplicantIncome  LoanAmount  Loan_Amount_Term  \
count       614.000000         614.000000  592.000000         600.00000   
mean       5403.459283        1621.245798  146.412162         342.00000   
std        6109.041673        2926.248369   85.587325          65.12041   
min         150.000000           0.000000    9.000000          12.00000   
25%        2877.500000           0.000000         NaN               NaN   
50%        3812.500000        1188.500000         NaN               NaN   
75%        5795.000000        2297.250000         NaN               NaN   
max       81000.000000       41667.000000  700.000000         480.00000   

       Credit_History  
count      564.000000  
mean         0.842199  
std          0.364878  
min          0.000000  
25%               NaN  
50%               NaN  
75%               NaN  
max          1.000000

1. Not all columns have corresponding numerical summary in the table. Why?
2. Compare the means vs. median. What does that imply?
3. 84.22% of the Applicants has credit history.

#### Distribution Analysis

#### Correlation Analysis

#### Predictive Modelling